nixos/modules/system/boot/networkd.nix

```diff
@@ -2470,6 +2470,7 @@ let
     address = mkOption {
       default = [ ];
       type = types.listOf types.str;
+      apply = warn "`systemd.network.networks.\"${name}\".address` is deprecated, use  `systemd.network.networks.\"${name}\".addresses` instead.";
```

Comment:
Went this route since `mkRenamedOptionModule` seems broken for submodules: https://github.com/NixOS/nixpkgs/issues/96006